test/builder.go

```diff
@@ -21,6 +27,36 @@ func Task(name, namespace string, ops ...func(*v1alpha1.Task)) *v1alpha1.Task {
 	return t
 }
 
+func TaskNamespace(namespace string) func(*v1alpha1.Task) {
```

Comment:
One thought I have is that it would be great to have some docs for the Builders - at the moment the best place for them would be https://github.com/knative/build-pipeline/blob/master/test/README.md#adding-integration-tests but I've been thinking for a while now that we should move that stuff into go docs instead, what do you think?

I think we should definitely include docstrings for each publicly exported function here anyway - and it would be super cool if we actually had some examples too!

We've even got a semi-decent start at https://godoc.org/github.com/knative/build-pipeline/test

Reply:
Definitely ! I didn't "yet" put godocs but we should definitely have some more for those builder